PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -1204,165 +1468,332 @@ TH1D* GetWeightsHist(const char* variable)
+  //               with abs path, for local files both abs and relative paths
+  //               are just fine.
   Bool_t bFileIsInAliEn = kFALSE;
+  Bool_t bFileIsInCCDB = kFALSE;
   if (TString(filePath).BeginsWith("/alice/cern.ch/")) {
     bFileIsInAliEn = kTRUE;
-  }
+  } else {
+    if (TString(filePath).BeginsWith("/alice-ccdb.cern.ch/")) {
+      bFileIsInCCDB = kTRUE;
+    } // else {
+  }   // if (TString(filePath).BeginsWith("/alice/cern.ch/")) {
 
-  // c) Handle the AliEn case:
-  TFile* oaFile = NULL; // file holding TObjArray with all labels
+  TFile *oaFile = NULL; // file holding TObjArray with all labels
   if (bFileIsInAliEn) {
-    TGrid* alien = TGrid::Connect("alien", gSystem->Getenv("USER"), "", "");
+    // c) Handle the AliEn case:
+    TGrid *alien = TGrid::Connect("alien", gSystem->Getenv("USER"), "", "");
     if (!alien) {
-      cout << __LINE__ << endl;
-      exit(1);
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
     }
     oaFile = TFile::Open(Form("alien://%s", filePath), "READ");
-  } else {
-    // d) Handle the local case:
+    if (!oaFile) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+    oaFile->GetObject(
+        "labels", oa); // TBI 20230530 hardcoded name of TObjArray is "labels",
+                       // perhaps I can do this also via Configurables? Keep in
+                       // sync with O2::TranslateASCIIintoObjArray
+    if (!oa) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+  } else if (bFileIsInCCDB) {
+
+    // d) Handle the CCDB case: Remember that here I do not access the file,
+    // instead directly object in that file.
+    //    Inspired by:
+    //    1. Discussion at:
+    //    https://alice-talk.web.cern.ch/t/access-to-lhc-filling-scheme/1073/17
+    //    2. See also:
+    //    https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/efficiencyGlobal.cxx
+    //                 https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/efficiencyPerRun.cxx
+
+    auto cm = o2::ccdb::BasicCCDBManager::instance();
+    cm.setURL("http://alice-ccdb.cern.ch");
+    // TH1F* hist = cm.get<TH1F>("Users/a/abilandz/test-2"); // this is the
+    // standard working example for histograms oa =
+    // cm.get<TObjArray>(TString(filePath).ReplaceAll("/alice-ccdb.cern.ch/","").Data());
+    // // TBI 20231004 doesn't work due to ownership problem, later in
+    // StoreLabelsInPlaceholder()
+    oa = (TObjArray *)(cm.get<TObjArray>(
+                           TString(filePath)
+                               .ReplaceAll("/alice-ccdb.cern.ch/", "")
+                               .Data()))
+             ->Clone(); // TBI 20231004 circumventing temporarily problem with
+                        // the ownership this way
```

Comment:
The same comment regarding the cloning of the CCDB object

Reply:
See the reply above.